crates/move-stackless-bytecode/src/function_target_pipeline.rs

```diff
@@ -409,15 +414,133 @@ impl FunctionTargetsHolder {
+                                        .display(func_env.symbol_pool())
+                                        .to_string();
+                                    env.diag(
+                                        Severity::Error,
+                                        &func_env.get_loc(),
+                                        &format!(
+                                            "Target struct '{}' not found in module '{}'",
+                                            struct_name, module_name
+                                        ),
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+            } else {
+                if func_env.get_name_str().contains("type_inv") {
+                    return;
+                } 
+
+                func_env
+                    .get_name_str()
+                    .strip_suffix("_inv")
+                    .map(|name: &str| {
+                        if let Some(struct_env) = func_env
+                            .module_env
+                            .find_struct(func_env.symbol_pool().make(name))
+                        {
+                            if self
+                                .datatype_invs
+                                .contains_left(&struct_env.get_qualified_id())
+                            {
+                                let env = func_env.module_env.env;
+                                env.diag(
+                                    Severity::Error,
+                                    &func_env.get_loc(),
+                                    &format!(
+                                        "Duplicate invariant declaration for struct: {}",
+                                        struct_env.get_full_name_str()
+                                    ),
+                                );
+                            } else {
+                                self.datatype_invs.insert(
+                                    struct_env.get_qualified_id(),
+                                    func_env.get_qualified_id(),
+                                );
+                            }
+                        } else {
+                            let env = func_env.module_env.env;
+                            let module_name = func_env
+                                .module_env
+                                .get_name()
+                                .display(func_env.symbol_pool())
+                                .to_string();
+                            env.diag(
+                                Severity::Error,
+                                &func_env.get_loc(),
+                                &format!(
+                                    "Target struct '{}' not found in module '{}'",
+                                    name, module_name
+                                ),
+                            );
+                        }
```

Comment:
once you have the `ModuleEnv` and the name of the struct, this is the same code as above. Let's refactor it to avoid duplication.